subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfigurationPublications.java

```diff
@@ -88,11 +88,11 @@ public void collectVariants(ConfigurationInternal.VariantVisitor visitor) {
         visitor.visitArtifacts(artifacts);
         PublishArtifactSet allArtifactSet = allArtifacts.getPublishArtifactSet();
         if (variants == null || variants.isEmpty() || !allArtifactSet.isEmpty()) {
-            visitor.visitOwnVariant(displayName, attributes.asImmutable(), allArtifactSet);
+            visitor.visitOwnVariant(displayName, attributes.asImmutable(), getCapabilities(), allArtifactSet);
```

Comment:
This change and the next one in the same file are where the information was lost: Outgoing variants of a configuration all share the capabilities of the configuration. There is effectively no way to store different capabilities information in the different outgoing variants.
Which means that the `VariantVisitor` needs to collect the capabilities of the outgoing configuration to populate the variants it creates.